contrib/exporters/secadvisor/mod/acc_rej_action.go

```diff
@@ -0,0 +1,218 @@
+	Saw2                     bool
+	Flow1                    *flow.Flow
+	Flow2                    *flow.Flow
+}
+
+type accRejAction struct {
+	// table of saved flows
+	accRejState map[string]*accRejFlowInfo
+	timeOfLastCleanup	int64
+}
+
+// NewAction - creates data structures needed for vpc Accept/Reject based on measurements flows at 2 points
+func NewAction(cfg *viper.Viper) (interface{}, error) {
+	accRejState := make(map[string]*accRejFlowInfo)
+	now := time.Now()
+	secs := now.Unix()
+	accRej := &accRejAction{
+		accRejState:  accRejState,
+		timeOfLastCleanup: secs,
+	}
+	return accRej, nil
+}
+
+// initAction - initialize internal data structure for this round
+func (t *accRejAction) initAction() {
+	// reset the entries in the table for this round
+	for _, ars := range t.accRejState {
+		ars.Saw1 = false
+		ars.Saw2 = false
+		ars.Status = FlowInfoOldFlow
+	}
+}
+
+// updateInfo - go over all the flows and fill in our internal data structure with relevant info
+func (t *accRejAction) updateInfo(fl []*flow.Flow) {
+	for _, f := range fl {
+		L3TrackingID := f.GetL3TrackingID()
+		flowInf, ok := t.accRejState[L3TrackingID]
+		if !ok {
+			// this flow is not yet in our table
+			t.accRejState[L3TrackingID] = &accRejFlowInfo{
+				UUID1:   f.UUID,
+				Status:  FlowInfoOneEntry,
+				Saw1:    true,
+				Flow1:   f,
+			}
+			continue
+		} else if flowInf.Status == FlowInfoOneEntry {
+			if flowInf.UUID1 != f.UUID {
+				flowInf.UUID2 = f.UUID
+				flowInf.Saw2 = true
+				flowInf.Flow2 = f
+				flowInf.Status = FlowInfoTwoEntries
+			} else {
+				// got duplicate from same flow; need to indicate reject
+				// or accumulate the stats until next time
+				flowInf.Saw1 = true
+				flowInf.Flow1 = f
+				continue
+			}
+		} else { // normal case
+			if flowInf.UUID1 == f.UUID {
+				flowInf.Flow1 = f
+				flowInf.Saw1 = true
+			} else {
+				flowInf.Flow2 = f
+				flowInf.Saw2 = true
+			}
+		}
+	}
+}
+
+// combineFlows - helper function to combine 2 seen flows into one, based on common minimums
+func combineFlows(f *accRejFlowInfo) *flow.Flow {
+	// perform shallow copy of flow and then update some fields
+	newF := *f.Flow1
+	var newMetricReported flow.FlowMetric
+	var newLastUpdateMetric flow.FlowMetric
+	// Note: Metric.Start should never change
+	newMetricReported.Start = f.Flow1.Metric.Start
+	// take byte and packet count as minimum of the 2 reports
+	newMetricReported.ABPackets = common.MinInt64(f.Flow1.Metric.ABPackets, f.Flow2.Metric.ABPackets)
+	newMetricReported.BAPackets = common.MinInt64(f.Flow1.Metric.BAPackets, f.Flow2.Metric.BAPackets)
+	newMetricReported.ABBytes = common.MinInt64(f.Flow1.Metric.ABBytes, f.Flow2.Metric.ABBytes)
+	newMetricReported.BABytes = common.MinInt64(f.Flow1.Metric.BABytes, f.Flow2.Metric.BABytes)
+
+	// if we are at the end of the flow, set Last time to latest of the times measured
+	if f.Flow1.FinishType != flow.FlowFinishType_NOT_FINISHED && f.Flow2.FinishType != flow.FlowFinishType_NOT_FINISHED {
```

Comment:
again this entire handling of Metrics just makes logic complex and output inconsistent - please remove

Reply:
Let's discuss this offline.